### ISSUE TITLE
Can I opera qml objects in go file?

### ISSUE THREAD

**Issue by fcying** [RadhiFadlillah/qamel#7]
c++ version:
```
  QObject *rect = object->findChild<QObject*>("rect");
  rect->setProperty("color", "red");
```

Does qamel provide this interface?

**Comment by RadhiFadlillah** [comment]
Hi @fcying, sorry for late response.

Unfortunately, `qamel` doesn't provide that interface, and while possible, I doubt I'll add it.

First, it needs many additional codes, which increase complexities. In fact, I've tried to make an API for this case, and it looks like this :

```go
viewer.SetObjectProperty("rect", "color", "red")
```

At first glance, it looks good and easy enough to do. The problem is, Go doesn't support generics, so we need to add several methods for each data type :

```go
viewer.SetObjectStringProperty("rect", "color", "red")
viewer.SetObjectBoolProperty("rect", "antialiasing", false)
viewer.SetObjectIntProperty("rect", "border.width", 3)
viewer.SetObjectFloatProperty("rect", "radius", 2.0)
...
```

In other hand, C++ supports generic, which make there are several properties that allow multiple type of value. For example, the value `gradient` in `rectangle` can be a string or an object. Sure we can use `interface{}` as placeholder, but then we will need to write codes for type assertion, which doesn't really fun to write.

Not to mention since `qamel` only supports basic primitive data type, we won't be able to set object as property. I think it's limiting and confusing, since some property can be set with this API, while other properties are not.

Next, according to Qt's [doc](https://doc.qt.io/qt-5/qtqml-cppintegration-interactqmlfromcpp.html), it said that :

> Although it is possible to access QML objects from C++ and manipulate them, it is not the recommended approach, except for testing and prototyping purposes. One of the strengths of QML and C++ integration is the ability to implement UIs in QML separate from the C++ logic and dataset backend, and this fails if the C++ side starts manipulating QML directly. Such an approach also makes changing the QML UI difficult without affecting its C++ counterpart.

---

In summary, after several times trying, I think it's too hard to do, and it makes the binding codes too complex for my taste. 

So, after long consideration, I don't think I will add it.

**Comment by fcying** [comment]
Thanks for your analysis
